packages/spacecat-shared-utils/src/log-wrapper.js

```diff
@@ -11,45 +11,46 @@
  */
 
 /**
- * Wraps the provided function to enhance the logging functionality by including the
- * `jobId` in each log statement.
+ * A wrapper function that enhances logging by appending the `jobId` to all log statements
+ * within the provided context. This allows better traceability and correlation of logs
+ * associated with a specific job.
  *
- * @param {function} fn - The original function that will be wrapped.
- * @returns {function(object, object): Promise<Response>} A new function that, when invoked,
- *  will automatically include the `jobId` in all log statements within the context.
+ * @param {function} fn - The original function to be wrapped.
+ * @returns {function(object, object): Promise<Response>} A wrapped function that includes
+ *  `jobId` in all log statements if `context.contextualLog` is used.
  */
 export function logWrapper(fn) {
   return async (message, context) => {
     const { log } = context;
 
     if (log) {
-      if (!message || !message.jobId) {
-        log.debug('Missing jobId, hence it will not be included in log messages.');
-        context.contextualLog = log;
+      if (message && 'jobId' in message) {
+        const { jobId } = message;
 
-        return fn(message, context);
+        // Enhance the log object to include jobId in all log statements
+        context.contextualLog = {
+          info: (...args) => {
+            log.info({ jobId, ...args });
```

Comment:
I have concerns about how this `...rest` and then `...spread` approach changes the resulting log format. Take the following simple function for example:

```
function logTest(...args) { console.log({jobId: "12345", ...args}); }
```

If I invoke the existing JS logger with multiple parameters, they are concatenated together to form a single log line:

```
> console.log("Here is my Log", "with something else (maybe a stacktrace)", "and an object:", {foo: "property"})
Here is my Log with something else (maybe a stacktrace) and an object: { foo: 'property' }
```

If I send the same params to our new logger, simplified in the `logTest` function above, I would be surprised that it produces a very different output:

```
> logTest("Here is my Log", "with something else (maybe a stacktrace)", "and an object:", {foo: "property"})
{
  '0': 'Here is my Log',
  '1': 'with something else (maybe a stacktrace)',
  '2': 'and an object:',
  '3': { foo: 'property' },
  jobId: '12345'
}
```

I think this is partially due to how we are taking the `...rest` and then spreading its values onto an object literal, which adds the index-like property names.

At this point, given no clear way to add a `key` (or `marker` in Splunk terms) to each entry, what if we simplified things to just include a prefix on each log entry containing the jobId? Modifying my simple logger above:

```
function logTest2(...args) { console.log('[jobId=12345]', ...args); }
```

Sample invocation and output:

```
> logTest2("Here is my Log", "with something else (maybe a stacktrace)", "and an object:", {foo: "property"})
[jobId=12345] Here is my Log with something else (maybe a stacktrace) and an object: { foo: 'property' }
```

This is much closer to the existing logging format expected from a logger, and does not have any degradation in terms of query-ability of the resulting log entry. WDTY?

Reply:
Sounds good to me. Just to clarify, if additional "markers" need to be added to the logger, would they appear like this: [jobId=12345, api-key=xyz, other-property=abc], or something like this: [jobId=12345] [api-key=xyz] [other-property=abc]? What are you envisioning?

Reply:
@blefebvre I have updated the "marker" format.

Reply:
> if additional "markers" need to be added to the logger

I think the format you described would work fine, to a point. Eventually it would be great to figure out how we can send custom "keys" to Coralogix, or if that is even available with the log API we are using.